### ISSUE TITLE
Can this function be supported?

### ISSUE THREAD

**Issue by chenenze** [hyperjumptech/grule-rule-engine#56]
```go
func (ab *AbstractModel) SumN(in ...int64) int64 {
	var sum int64
	for _, v := range in {
		sum += v
	}
	return sum
}
```

**Comment by newm4n** [comment]
Hi chenenze

Golang's variadic argument is not supported

**Comment by chenenze** [comment]
thx, can you tell me your considerations?

**Comment by newm4n** [comment]
There are 2 reason as this feature are not (yet) implemented:

1. While it's always a good idea to introduce this variadic argument to enrich the grule language, the original aim of grule is to make a simple language for rule-engine similar to Grools. We are not aiming to "match" golang capability as a language. There are other similar feature been requested (eg. multiple return values), but we refrain to implement them. 
2. If these feature gains enough attentions, requests, we might going to implement them. But still it's going to be some one to do it. Kind contribution is greatly appreciated.

I hope this answers.

**Comment by chenenze** [comment]
Thanks for your answers，I am investigating in order to use it in a production environment，Do you know any company uses this project in a production environment？

**Comment by newm4n** [comment]
To be honest, I don't know who have use this software in Production. This is an open-source software and provided as is. I have not seen any feedback of their use in production. 

While we always put effort to have it "Production" grade, it's still provided free and as is. if you could help us to test it more and more, it would be a great contribution.

**Comment by niallnsec** [comment]
Hi,

I just submitted a PR which would add support for variadic functions as this is also a feature I had need of.

The project I am currently using grule in is intended to go into production and is currently in beta. So far we have been really happy with performance and stability. There are a number of features we found necessary to our specific use case that probably don't fit well for the primary package so we are currently working on a fork with the following changes:
 - Event Bus removed - We are expecting the rule engine to be executed thousands of times a second and are not using the event bus, so removing it made sense to save time/allocations.
 - Built in Complete() function for use in the then expression to terminate execution of the engine after a specific rule match.
 - Underscores allowed in rule names.
 - (WIP) Support for nested function calls as per issue #54 
Apart from those changes everything else is the same. I will of course continue to provide updates as we continue our testing, as well as any improvements/fixes we make which fit with this library.
